authentication/provider/service/authentication_provider_service.go

```diff
@@ -0,0 +1,645 @@
+				}
+				userNotApprovedRedirectURL, err := rest.AddParam(userNotApprovedRedirectURL, "status", status)
+				if err != nil {
+					log.Error(ctx, map[string]interface{}{"err": err}, "failed to add a status param to the redirect URL")
+					return nil, nil, err
+				}
+				log.Debug(ctx, map[string]interface{}{
+					"user_not_approved_redirect_url": userNotApprovedRedirectURL,
+				}, "user not approved; redirecting to registration app")
+				return &userNotApprovedRedirectURL, nil, nil
+			}
+			return nil, nil, autherrors.NewUnauthorizedError(err.Error())
+		}
+		return nil, nil, err
+	}
+
+	if identity.User.Deprovisioned {
+		log.Warn(ctx, map[string]interface{}{
+			"identity_id": identity.ID,
+			"user_name":   identity.Username,
+		}, "deprovisioned user tried to login")
+		return nil, nil, autherrors.NewUnauthorizedError("unauthorized access")
+	}
+
+	log.Debug(ctx, map[string]interface{}{
+		"referrerURL": referrerURL.String(),
+		"user_name":   identity.Username,
+	}, "local user created/updated")
+
+	// Generate a new user token instead of using the original oauth provider token
+	userToken, err := tokenManager.GenerateUserTokenForIdentity(ctx, *identity, false)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{"err": err, "identity_id": identity.ID.String()}, "failed to generate token for user")
+		return nil, nil, err
+	}
+
+	err = encodeToken(ctx, referrerURL, userToken, apiClient)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to encode token")
+		redirectTo := referrerURL.String() + err.Error()
+		return &redirectTo, nil, autherrors.NewInternalError(ctx, err)
+	}
+	log.Debug(ctx, map[string]interface{}{
+		"referrerURL": referrerURL.String(),
+		"user_name":   identity.Username,
+	}, "token encoded")
+
+	redirectTo := referrerURL.String()
+	return &redirectTo, userToken, nil
+}
+
+// UpdateIdentityFromExistingIdentityInfo update identity if exists using profile information and returns updated identity.
+func (s *authenticationProviderServiceImpl) UpdateIdentityUsingUserInfoEndPoint(ctx context.Context, accessToken string) (*account.Identity, error) {
+	userProfile, err := s.Factories().IdentityProviderFactory().NewIdentityProvider(ctx, s.config).Profile(ctx, oauth2.Token{AccessToken: accessToken})
+
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"token": accessToken,
+			"err":   err,
+		}, "unable to get user profile")
+		return nil, errors.New("unable to get user profile " + err.Error())
+	}
+
+	identity := &account.Identity{}
+
+	identities, err := s.Repositories().Identities().Query(account.IdentityFilterByUsername(userProfile.Username), account.IdentityWithUser())
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "unable to query for an identity by username")
+		return nil, errs.Wrapf(err, "error during querying for an identity by ID")
+	}
+
+	if len(identities) == 0 {
```

Comment:
same here, not sure this needs to be checked

Reply:
As above, it looks like this check is actually necessary.